codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/GoMiddlewareGenerator.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.go.codegen;
+
+import java.util.function.BiConsumer;
+
+import software.amazon.smithy.codegen.core.Symbol;
+import software.amazon.smithy.codegen.core.SymbolReference;
+
+/*
+ * Helper for generating Smithy Go middleware
+ */
+public class GoMiddlewareGenerator {
+    private static final SymbolReference CONTEXT_TYPE;
+    private static final SymbolReference METADATA_TYPE;
+
+    private final String identifier;
+    private final String handleMethodName;
```

Comment:
Curious if anyone has thoughts on this also being defined as a symbol with a relevant getter method. Reason being is the expected pattern is that the implementation of the handler body calls the `next.%handleMethodName(ctx, in)`. Maybe this makes more sense as a symbol?